### ISSUE TITLE
Implement month and era codes

### ISSUE THREAD

**Issue by sffc** [unicode-org/icu4x#486]
The syntax for month and era codes is being discussed in #355.  I implemented a placeholder in #445.  This ticket is to follow up and properly wire month codes throughout the data model and document them appropriately.

**Comment by sffc** [comment]
I will do this along with #257.

**Comment by sffc** [comment]
Blocked on #519

**Comment by sffc** [comment]
Blocked on #493

**Comment by Manishearth** [comment]
Reassigning to self since this is a part of https://github.com/unicode-org/icu4x/issues/1115 and https://github.com/unicode-org/icu4x/issues/1116

I plan to do the era code part for now, not month codes (not yet, anyway)

**Comment by Manishearth** [comment]
Split it out into a separate issue since month codes will likely best be paired with a calendar that needs them

https://github.com/unicode-org/icu4x/issues/1312

**Comment by gregtatum** [comment]
I am unclear what this is exactly. @Manishearth @sffc would you mind clarifying and pointing to UTS 35 for this? It's marked with the C-datetime label, and we're triaging the DateTimeFormat component.

**Comment by Manishearth** [comment]
This isn't a UTS 35 thing, UTS 35 has a month naming scheme that's somewhat inadequate: fine for data, but not for user-visible APIs.

**Comment by Manishearth** [comment]
I think the general plan will be to move over to using month codes as data indices. `Month` may also contain an ordinal month, but not _necessarily_.

**Comment by sffc** [comment]
@macchiati @pedberg-icu For calendars that insert months into the middle of the year, if a numeric-only month format is requested, is it acceptable to use the ordinal month number?

**Comment by Manishearth** [comment]
I've filed https://github.com/unicode-org/icu4x/issues/2066 for leap month things and will be referencing it in code wehre leap months need to be handled, because I'm wary of writing code that can't be tested.
